packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -126,6 +126,12 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 					if (typeof query.index === 'undefined') {
 						throw new Error(`Requests for hoisted scripts must include an index`);
 					}
+					// HMR hoisted script only exists to make them appear in the module graph.
+					if(opts?.ssr) {
+						return {
+							code: ''
```

Comment:
nit: can this be a comment like `/* client hoisted script, empty in SSR */`? Bonus if it can also include the file path or some other useful debug info.